### PR TITLE
attempted refactoring of main.py

### PR DESCRIPTION
**Pull request by frankdai320**
I couldn't be bothered to test it :)

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -46,6 +46,243 @@ def main():
...
+    current_chat = message['chat']['id']
+    data = {'chat_id': current_chat,
+            'text': 'Hello World!'}
+    tg.send_message(data)
+
+
+bot_commands["/helloworld"] = helloworld
+
+
+def source(message):
```
Comment (jarhill0) [inline main.py:64]: `current_chat` is not defined

File: main.py [hunk 2]
```diff
@@ -46,6 +46,243 @@ def main():
...
+
+def start(message):
+    user = message['from']['first_name']
+    bot_message = 'Hello %s! I am A Bot.' % user
+    bot_message += "\n\nHere's a list of my commands:" \
+                   "\n/redditposts [subreddit] - List 5 hot posts from /r/[subreddit]" \
+                   "\n/reddit [shortlink] - Get info on the linked Reddit post." \
+                   "\n/wa - Follow with a query to get information from WolframAlpha" \
+                   "\n/nextlaunch - Get information on the next SpaceX launch"
+    data = {'chat_id': current_chat,
```
Comment (jarhill0) [inline main.py:83]: again, `current_chat` is not defined.

File: main.py (hunk 1 above)
Comment (frankdai320) [reply main.py:64]: wut I defined it right there

File: main.py [hunk 3]
```diff
@@ -46,6 +46,243 @@ def main():
...
+            'text': bot_message,}
+    tg.send_message(data)
+
+
+bot_commands["/start"] = start
+
+
+def help(message):
+    data = {'chat_id': current_chat,
```
Comment (jarhill0) [inline main.py]: `current_chat` isn't defined

File: main.py (hunk 1 above)
Comment (frankdai320) [reply main.py:64]: oh I see

File: main.py [hunk 4]
```diff
@@ -46,6 +46,243 @@ def main():
...
+                    "\n/wa - Follow with a query to get information from WolframAlpha"
+                    "\n/nextlaunch - Get information on the next SpaceX launch",}
+    tg.send_message(data)
+
+
+bot_commands["/help"] = help
+
+
+def settings(message):
+    data = {'chat_id': current_chat,
```
Comment (jarhill0) [inline main.py]: `current_chat` isn't defined

File: main.py [hunk 5]
```diff
@@ -46,6 +46,243 @@ def main():
...
+    message_text = message.get('text', None)
+    command_block = message_text[message_text.index('/redditposts'):]
+    try:
+        subreddit = command_block.split(' ')[1]
+    except IndexError:
+        bot_message = 'Specify a subreddit after /redditposts (e.g. /redditposts funny)'
+    else:
+        bot_message, posts_dict = reddit.hot_posts(subreddit, 5)
+    finally:
+        data = {'chat_id': current_chat,
```
Comment (jarhill0) [inline main.py]: `current_chat` isn't defined

File: main.py [hunk 6]
```diff
@@ -46,6 +46,243 @@ def main():
...
+
+
+bot_commands["/redditposts"] = redditposts
+
+
+def stop(message):
+    username = message['from'].get('username', None)
+    if (username == config.owner_un or username in config.owner_uns) and time.time() - \
+            message[
+                'date'] < 15:
```
Comment (jarhill0) [inline main.py]: that sure is some goofy formatting

File: main.py [hunk 7]
```diff
@@ -46,6 +46,243 @@ def main():
...
+        sys.exit()
```
Comment (jarhill0) [inline main.py:155]: bug in my code: this is all wrapped in a try-except, so I'm pretty sure the `sys.exit()` call fails every time.

Comment (frankdai320) [reply main.py:155]: no it doesn't

**Comment by jarhill0** [comment]
It seems to work!

**Pull request merged**